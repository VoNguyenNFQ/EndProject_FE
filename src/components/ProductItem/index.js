import React from 'react';
import { Link } from 'react-router-dom';
import { formatMoney } from 'utils/formatNumber'
import 'assets/css/main.css';

const ProductItem = ({ product}) => {
    const { id, name, price, gallery } = product;
    //417
    return (
        <Link to={`/product-list/${id}`}>
            <div id={"product-" + id} className='flex flex-col items-center justify-center w-full max-w-lg mx-auto'>
            <div class="img-hover-zoom">
                <img src={gallery[0]} alt={name} className={`h-[417px] object-cover w-full rounded-md hover:shadow-[0_0_8px_1px_#ff529c]`} />
            </div>
                <h4 className='mt-2 text-lg font-normal text-gray-700 text-center truncate w-full'>{name}</h4>
                <p className='text-pink-400 font-bold'>{formatMoney(price)}</p>
            </div>
        </Link>
    );
};

export default ProductItem;