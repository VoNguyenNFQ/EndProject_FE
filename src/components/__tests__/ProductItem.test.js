import { faker } from '@faker-js/faker';
import ProductItem from 'components/ProductItem';
import ReactDOM from 'react-dom';

describe("Unit Test Component Product Item", () => {
    let container;

    beforeEach(() => {
        container = document.createElement('div')
        document.body.appendChild(container)
    })

    afterEach(() => {
        document.body.removeChild(container)
        container = null;
    })

    it("Display product Item", () => {
        const product = {
            id: faker.datatype.number,
            name: faker.commerce.productName,
            price: faker.commerce.price,
            gallery: faker.image
        }

        act(() => {
            ReactDOM.render(<ProductItem product={product} />, container);
        })

        
    })
})