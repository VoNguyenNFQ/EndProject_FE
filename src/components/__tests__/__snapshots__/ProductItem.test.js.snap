// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test Component Product Item Display product Item 1`] = `
<h4
  className="mt-2 text-lg font-normal text-gray-700 truncate w-full"
>
  Giày cao gót
</h4>
`;
